### PR TITLE
nixos/getty: add services.getty.loginOptions

### PR DESCRIPTION
**Pull request by alyssais**
This corresponds to agetty's --login-options argument.

With this change, I can set

    services.getty.autologinUser = "qyliss";
    services.getty.loginOptions = "-- \\u";

and have my username prefilled, but with my password still
required (unlike the normal autologinUser behaviour).

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by alyssais** [comment]
@ofborg test login

**Comment by alyssais** [comment]
@ofborg test login

File: nixos/modules/services/ttys/getty.nix [hunk 1]
```diff
@@ -3,9 +3,19 @@
...
+    "--autologin" cfg.autologinUser
+  ] ++ optionals (cfg.loginOptions != null) [
+    "--login-options" cfg.loginOptions
+  ];
 
-  autologinArg = optionalString (config.services.getty.autologinUser != null) "--autologin ${config.services.getty.autologinUser}";
-  gettyCmd = extraArgs: "@${pkgs.util-linux}/sbin/agetty agetty --login-program ${pkgs.shadow}/bin/login ${autologinArg} ${extraArgs}";
+  gettyCmd = extraArgs:
+    "@${pkgs.util-linux}/sbin/agetty agetty ${escapeShellArgs loginArgs} "
+      + extraArgs;
```
Comment (cole-h) [inline nixos/modules/services/ttys/getty.nix:18]: A little curious why you opted for `+ extraArgs` instead of `${extraArgs}"` here.

**Review by cole-h — APPROVED** [review]
Diff LGTM.

Pretty neat, I think I'll do the same if (when) this gets merged.

File: nixos/modules/services/ttys/getty.nix (hunk 1 above)
Comment (alyssais) [reply nixos/modules/services/ttys/getty.nix:18]: The line was getting too long for my liking.

**Pull request merged**